### PR TITLE
added checkAlliance() method

### PR DESCRIPTION
**Pull request by sammyredha**

### CODE REVIEW

**Review by Sid220 — APPROVED** [review]
Looks good!

File: src/main/java/frc/robot/Robot.java [hunk 1]
```diff
@@ -481,6 +487,18 @@ public void testPeriodic() {}
...
 
+  public void checkAlliance() {
+    Alliance checkedAlliance = DriverStation.getAlliance();
+
+    if (DriverStation.isDSAttached() && checkedAlliance != currentAlliance) {
+      currentAlliance = checkedAlliance;
+
+      goClosestGrid = new GoClosestGrid();
+      autoChooser.addDefaultOption("TwoBridgeOver", new TwoCargoOver());
+      autoChooser.addOption("TwoBridgeUnder", new TwoCargoUnder());
```
Comment (tervay) [inline src/main/java/frc/robot/Robot.java:498]: This is the same code as lines 153-154, can we move these into a `buildAutoChooser()` method?

**Review by tervay — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/Robot.java (hunk 1 above)
Comment (sammyredha) [reply src/main/java/frc/robot/Robot.java:498]: good point, ill do that now

**Comment by tervay** [comment]
Make sure to put `checkAlliance()` in `disabledPeriodic()` so that we can continuously check which alliance we are while we are waiting for the match to start.

**Pull request merged**